File: typescript/src/creational/builder/dto/Employee.ts
import { Phone } from "./Phone";
import { Address } from "./Address";
import { Contact } from "./Contact";

export interface Builder<T> {
    build(): T; 
}

export enum GENDER {
    MALE = "Male",
    FEMALE = "Female",
    OTHER = "Other"
}

export class Employee {
    private name: string
    private age: number
    private gender: GENDER
    private address: Address
    private phones: Array<Phone>
    private contacts: Array<Contact>

    constructor(name: string, age: number, gender: GENDER, address: Address, phones: Array<Phone>, contacts: Array<Contact> ){
        this.name = name
        this.age = age
        this.gender = gender
        this.address = address
        this.phones = phones
        this.contacts = contacts
    }

    toString(): string {
        return (
            `Employee\n` +
            `Name: ${this.name}\n` +
            `Age: ${this.age}\n` +
            `Gender: ${this.gender}\n` +
            `Address: ${this.address.toString()}\n` +
            `Phones: ${this.phones.map(phone => phone.toString()).join(", ")}\n` +
            `Contacts: ${this.contacts.map(contact => contact.toString()).join(", ")}`
    );
    }

}

File: c#/Creational/Builder/Dto/Employee.cs
using Dto.Address;
using Dto.Contact;
using Dto.Phone;

namespace Dto.Employee
{
    public enum Gender
    {
        Male,
        Female,
        Other
    }
    public class EmployeeDto
    {
        private string name;
        private int age;
        private Gender gender;
        private AddressDto address;
        private List<PhoneDto> phones = new List<PhoneDto>();
        private List<ContactDto> contacts = new List<ContactDto>();

        public EmployeeDto(string name, int age, Gender gender, AddressDto address, List<PhoneDto> phones, List<ContactDto> contacts)
        {
            this.name = name;
            this.age = age;
            this.gender = gender;
            this.address = address;
            this.phones = phones;
            this.contacts = contacts;
        }

        public override string ToString()
        {
            var phonesStr = string.Join(", ", phones.Select(p => p.ToString()));
            var contactsStr = string.Join(", ", contacts.Select(c => c.ToString()));
            return $"Employee \n Name: {name} \n Age: {age} \n Gender: {gender} \n Address: {address} \n Phones: {phonesStr} \n Contacts: {contactsStr}";
        }

    }
}